bookwyrm/models/fields.py

```diff
@@ -387,7 +387,8 @@ def set_activity_from_field(self, activity, instance):
         activity[key] = formatted
 
     def field_to_activity(self, value, alt=None):
-        return image_serializer(value, alt)
+        url = self.get_absolute_url(value)
+        return activitypub.Document(url=url, name=alt)
```

Comment:
Okay I think I figured out the issue - `image_serializer` was only creating the `activitypub.Document` object if the image url exists. In this function, it's created even if the image is `None`. If you check whether the url is a falsey value and return None, I think that'll fix the issues.